client/src/containers/SubmitCode.js

```diff
@@ -1,62 +1,93 @@
 import React, { PropTypes, Component } from 'react';
-import { submitCode } from '../actions/code';
+import { getUserInfo } from '../actions/user.js';
 import { connect } from 'react-redux';
 
 class SubmitCode extends Component {
 	constructor(props) {
 		super(props);
-		this.state = {
-			value: '(e.g. CS 242 - Chess)', 
-			language: {value: 'Java'}
-		};
 		// React components in ES6 no long autobinds this to nonReact methods
-		this.handleChange = this.handleChange.bind(this);
-		this.handleSelect = this.handleSelect.bind(this);
-		this.handleSubmit = this.handleSubmit.bind(this);
+		this.handleTitleChange = this.handleTitleChange.bind(this);
+		this.handleLanguageSelect = this.handleLanguageSelect.bind(this);
+		this.handleContentChange = this.handleContentChange.bind(this);
+		this.routeToSubmitCodeSmells = this.routeToSubmitCodeSmells.bind(this);
+		this.state = {
+			content: ""
+		}
+	}
+
+	componentDidMount() {
+        const { dispatch } = this.props;
+        dispatch(getUserInfo());
+    }
+
+	handleTitleChange(evt) {
+		const { code } = this.props;
+		code.title = evt.target.value;
 	}
-	handleChange(evt) {
-		this.setState({
-			value: evt.target.value
-		});
+
+	handleLanguageSelect(evt) {
+		const { code } = this.props;
+		code.language = evt.target.value;
 	}
-	handleSelect(evt) {
-		this.setState({
-			language: evt.target.value
-		});
+
+	handleContentChange(evt) {
+		const { code } = this.props;
+		// console.log("CHANGING CONTENT " + evt.target.value);
+		code.content = evt.target.value;
 	}
-    handleSubmit(evt) {
-        submitCode(this.state.value, this.state.language);
+
+    routeToSubmitCodeSmells() {
+    	const { dispatch, user, code } = this.props;
+        if (code.title === "") {
+        	console.log("You need to input title!");
+        } else if (code.language === "") {
+        	console.log("You need to input language!");
+        } else if (code.content === "") {
+        	console.log("You need to input content!");
+        } else {
+        	localStorage.setItem("userid", user.id);
+        	localStorage.setItem("code", JSON.stringify(code));
+        	this.context.history.pushState(null, '/submitSmells');
```

Comment:
This is nitpicky but you should just be using dispatch(pushState) because I'm not entirely sure if doing it this way saves the history so it won't go back if you navigate back.